File: examples/athletic-analyst-py/scripts/discover_tools.py
"""Discover and list tools exposed by the Vytallink MCP server."""

import asyncio
import json
import os
import sys
from pathlib import Path

# Allow running as `python scripts/discover_tools.py` from athletic-analyst-py/
sys.path.insert(0, str(Path(__file__).resolve().parent.parent))

from mcp import ClientSession, StdioServerParameters
from mcp.client.stdio import stdio_client

from src.mcp_bridge import build_mcp_environment


async def main() -> None:
    print("Connecting to Vytallink MCP server (deployed backend)...")

    server_params = StdioServerParameters(
        command="npx",
        args=["@xmartlabs/vytallink-mcp-server"],
        env=build_mcp_environment(),
    )

    async with stdio_client(server_params) as (read_stream, write_stream):
        async with ClientSession(read_stream, write_stream) as session:
            await session.initialize()
            result = await session.list_tools()
            tools = result.tools

            print(f"Connected. Found {len(tools)} tools:\n")
            for t in tools:
                desc = (t.description or "")[:80]
                print(f"  - {t.name}: {desc}...")

            docs_dir = Path(__file__).resolve().parent.parent.parent / "docs"
            docs_dir.mkdir(parents=True, exist_ok=True)
            output_path = docs_dir / "vytallink-tools.json"

            tools_data = [
                {
                    "name": t.name,
                    "description": t.description,
                    "inputSchema": t.inputSchema,
                }
                for t in tools
            ]
            output_path.write_text(json.dumps(tools_data, indent=2))
            print(f"\nSaved to {output_path}")


if __name__ == "__main__":
    asyncio.run(main())

File: examples/athletic-analyst-ts/scripts/discover-tools.ts
import { Client } from "@modelcontextprotocol/sdk/client/index.js";
import { StdioClientTransport } from "@modelcontextprotocol/sdk/client/stdio.js";
import { writeFileSync, mkdirSync } from "node:fs";
import { resolve, dirname } from "node:path";
import { tmpdir } from "node:os";
import { buildMcpEnvironment } from "../src/mcp-bridge.js";

function buildOutputPath(): string {
  const customPath = process.env.VYTALLINK_DISCOVERY_OUTPUT;
  if (customPath && customPath.trim().length > 0) {
    return resolve(customPath);
  }

  return resolve(tmpdir(), "vytallink-agent-examples", "vytallink-tools.json");
}

console.log("Connecting to Vytallink MCP server (deployed backend)...");

const transport = new StdioClientTransport({
  command: "npx",
  args: ["@xmartlabs/vytallink-mcp-server"],
  env: buildMcpEnvironment(),
});

const client = new Client({ name: "discovery", version: "1.0.0" });

try {
  await client.connect(transport);
  console.log("Connected. Listing tools...");

  const { tools } = await client.listTools();
  console.log(`Found ${tools.length} tools:\n`);
  tools.forEach((t) => console.log(`  - ${t.name}: ${t.description?.slice(0, 80)}...`));

  const outputPath = buildOutputPath();
  mkdirSync(dirname(outputPath), { recursive: true });
  writeFileSync(outputPath, JSON.stringify(tools, null, 2));
  console.log(`\nSaved to ${outputPath}`);
} finally {
  await client.close();
}
